src/Splunk.OpenTelemetry.AutoInstrumentation/Logging/FileSink.cs

```diff
@@ -0,0 +1,79 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated />
+// This comment is here to prevent StyleCop from analyzing a file originally from Serilog.
+//------------------------------------------------------------------------------
+
+// Copyright 2013-2019 Serilog Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.IO;
+using System.Text;
+using OpenTelemetry.AutoInstrumentation.Logging;
+using Splunk.OpenTelemetry.AutoInstrumentation.Logging;
+
+namespace OpenTelemetry.AutoInstrumentation.Logging
+{
+    internal sealed class FileSink : ISink, IDisposable
+    {
+        readonly TextWriter _output;
+        readonly FileStream _underlyingStream;
+        readonly WriteCountingStream _countingStreamWrapper;
+        readonly object _syncRoot = new object();
+        static readonly long FileSizeLimitBytes = 10 * 1024 * 1024;
+
+        public FileSink(string path, Encoding encoding = null)
+        {
+            if (path == null) throw new ArgumentNullException(nameof(path));
+
+            var directory = Path.GetDirectoryName(path);
+            if (!string.IsNullOrWhiteSpace(directory) && !Directory.Exists(directory))
+            {
+                Directory.CreateDirectory(directory);
+            }
+
+            _underlyingStream = System.IO.File.Open(path, FileMode.Append, FileAccess.Write, FileShare.Read);
```

Comment:
Does it really work properly? Can we open the same file for `FileAccess.Write, FileShare.Read` twice in the same process?